server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2278,24 +2278,9 @@ public void syncRetentionLeases() {
         verifyNotClosed();
         replicationTracker.renewPeerRecoveryRetentionLeases();
         final Tuple<Boolean, RetentionLeases> retentionLeases = getRetentionLeases(true);
-        if (retentionLeases.v1()) {
```

Comment:
With this change, I think we no longer need to return a tuple in `getRetentionLeases(true)`?

Reply:
Thanks Nhat, I will fix that in a follow-up, since it requires non-trivial test changes.